src/System.Net.Http/tests/StressTests/HttpStress/ClientOperations.cs

```diff
@@ -149,21 +179,55 @@ public static class ClientOperations
                     using (var req = new HttpRequestMessage(HttpMethod.Get, "/headers") { Version = httpVersion })
                     using (HttpResponseMessage m = await ctx.SendAsync(req))
                     {
-                        ValidateResponse(m, httpVersion);
+                        ValidateHttpVersion(m, httpVersion);
+                        ValidateStatusCode(m);
                         ValidateContent(ctx.ContentSource, await m.Content.ReadAsStringAsync());
                     }
                 }),
 
+                ("GET Echo Headers",
+                async ctx =>
+                {
+                    Version httpVersion = ctx.GetRandomHttpVersion();
+
+                    using (var req = new HttpRequestMessage(HttpMethod.Get, "/echoHeaders") { Version = httpVersion })
+                    {
+                        ctx.PopulateWithRandomHeaders(req.Headers);
+
+                        using (HttpResponseMessage res = await ctx.SendAsync(req))
+                        {
+                            ValidateHttpVersion(res, httpVersion);
+                            ValidateStatusCode(res);
+
+                            // Validate that response headers are being echoed
+                            foreach (var reqHeader in req.Headers)
+                            {
+                                if (!res.Headers.TryGetValues(reqHeader.Key, out var values))
```

Comment:
Is `TryGetValues` case-insensitive? You've got upper-case characters in `alphaNumeric` and for HTTP2 these should all get lower-cased prior to sending over the wire.

Reply:
Yes it is.